### PR TITLE
Initial www-dev playbook

### PR DESCRIPTION
**Pull request by kennethgillen**
Development of a new web server for OME website migration.

Private config in https://github.com/openmicroscopy/management_tools/pull/474

### CODE REVIEW

File: ansible/server-state-playbooks/www-dev/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "centos/7"
+  config.vm.provider "virtualbox" do |vb|
+  config.vm.network "forwarded_port", guest: 80, host: 8080
+  config.vm.network "forwarded_port", guest: 4064, host: 4064
```
Comment (sbesson) [inline ansible/server-state-playbooks/www-dev/Vagrantfile]: Probably not required here

File: ansible/server-state-playbooks/www-dev/www-dev.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        owner="{{ item.value.owner }}"
+        group="{{ item.value.group }}"
+        mode="{{ item.value.mode }}"
+      with_dict: "{{ nginx_ssl_cert_files }}"
+      no_log: true
+
+    - name: NGINX config | www-dev.openmicroscopy.org
+      become: yes
+      template:
+        src: templates/www-dev.openmicroscopy.org.conf.j2
```
Comment (sbesson) [inline ansible/server-state-playbooks/www-dev/www-dev.yml]: Are the templates meant to be committed as part of this playbook as well?

Comment (kennethgillen) [inline ansible/server-state-playbooks/www-dev/www-dev.yml]: Yes, I had intended to add the config here via the template too. Missed in amongst the other (intentionally) not-included files. Thanks for noticing.

File: ansible/server-state-playbooks/www-dev/Vagrantfile (hunk 1 above)
Comment (kennethgillen) [inline ansible/server-state-playbooks/www-dev/Vagrantfile]: Addressed in https://github.com/openmicroscopy/infrastructure/pull/281/commits/c3845609de90d79fb08b2b3858e63b3e1cfcf811

**Comment by kennethgillen** [comment]
Thanks for review @sbesson  - I am planning to remove the ON_HOLD label once the relevant config is able to be merged, i.e. after the removal of vault.

**Comment by sbesson** [comment]
A couple of additions proposed in https://github.com/kennethgillen/infrastructure/pull/1 to match the current gem requirements of www.openmicroscopy.org (incl. . There are still a few gotchas but I think we are close to a state where we can merge this playbook as a first working pass and start working on the nginx configuration.

**Comment by kennethgillen** [comment]
Thanks, Seb. Not sure how we started NGINX before.. but good to have it in now.

**Comment by sbesson** [comment]
With the last commit, tried to the deploy the entire website from a freshly snapshoted VM using `ansible-playbook --ask-become-pass server-state-playbooks/www-dev/www-dev.yml`. Barring the issue discussed below, everything worked as expected with the static content being served by nginx as expected.

Merging this alongside with the private PR containing the variables to allow the work on the Nginx configuration files (SSL + redirects work). On the Jekyll front, the main issue is that the `bundle install` step systematically fails with ` No such file or directory` error forcing the playbook to be executed twice for every code change. Merging this playbook as it is also allows to investigate this issue separately.

**Pull request merged**